pkg/operator/controller/ingress/controller.go

```diff
@@ -611,7 +611,11 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 	}
 
 	if err := r.syncIngressControllerStatus(ci, deployment, lbService, operandEvents.Items, wildcardRecord, dnsConfig); err != nil {
-		errs = append(errs, fmt.Errorf("failed to sync ingresscontroller status: %v", err))
+		syncErr := fmt.Errorf("failed to sync ingresscontroller status: %v", err)
```

Comment:
Would using error wrapping remove the need to put this sort of code everywhere?

Reply:
Because then we could unwrap the errors once on the outside and check to see if any of the errors in the chain are retryable

Reply:
I think so, but I am going to change this PR not to wrap `err` at all because `syncIngressControllerStatus` decorates its errors sufficient, and we don't want to log "failed to sync ingresscontroller status" if the error is a retryable error (in which case there was no failure to update status).